All my updates

evandrocoan opened the pull request.

evandrocoan committed:
Moved the settings documentation from readme.md to its file.

evandrocoan committed:
Fixed evandrocoan/ITE#84 TypeError: 'NoneType' object does not

support item assignment.

evandrocoan committed:
Fixed variable naming and function blocks new line spacing.

evandrocoan committed:
Converted WordCount.py indentation to spaces.

evandrocoan committed:
Fixed the line count status being erased a few seconds after

opening the view/file.

evandrocoan committed:
Removed dead code on WordCount.py

evandrocoan committed:
Rewrote almost everything on WordCount.py and deprecated the

settings:

// An array of syntax names that WordCount should run on. Example:
// ["Plain text", "Markdown"] If the array is empty, like it is by default,
// WordCount will run on any syntax.
"whitelist_syntaxes": [],

// An array of syntax names that WordCount should not run on.
// Example: ["Plain text", "Markdown"]
// If the array is empty, like it is by default, WordCount will run on any syntax.
"blacklist_syntaxes": [],

// Display the count of words found on current line.
"enable_line_word_count": false,

// Display the count of characters found on current line.
"enable_line_char_count": false,

evandrocoan committed:
Created the setting "enable_count_words": true, to display the

number of words in file.

evandrocoan committed:
Removed deprecated super constructor call on WordCountView class.

evandrocoan committed:
Moved the WordCount.py setup functions near the top of the file.

evandrocoan committed:
Implemented word count when there are selections only.

evandrocoan committed:
Fixed WordCount.py views clean up and removed deprecated settings.

evandrocoan committed:
Fixed the main thread not closing properly when unloading the

plugin WordCount.py

evandrocoan committed:
The word_count_loop() thread stop condition should be placed before

the doCountion() call because it will allow the process to stop
right after the g_is_already_running sentinel variable is set to
false.

evandrocoan committed:
Deprecated if countView guard on doCounting() as it should not be

None anymore.

evandrocoan committed:
Changed the time rule to 3.0 seconds with penality of 100 times the

overused time after 0.01 seconds.

evandrocoan committed:
Improved the selections word counting by not copying all the

selections strings into one big string.

titoBouzout commented:
Hi! I notice given the change set provided that you care enough for it. I can understand the meaning of the commits, but can you please make a list of which feature is no longer supported and what new features has? 

In your version when I open a 70Mb file the count get stuck at 6millons words and does not change when I select stuff.  

I dont know how much motivation you have to merge this, I just would like to suggest to not deprecate features unless you have a good reason for it, all of the features are there because someone asked for it.

evandrocoan commented:
> I just would like to suggest to not deprecate features unless you have a good reason for it

Hi! Thanks for looking into it, I had removed some stuff for 2 reasons (First Reason) is that I was having a hard time into making it work correctly, then I started refactoring and removing some stuff which I did not liked there because (Second Reason) they seemed some very expensive for performance.

> can you please make a list of which feature is no longer supported and what new features has?

I think I did not removed much stuff, because when I remove something from the code, I also take it off from the `.sublime-settings` file, then looking at this pull request changes, I noticed the only stuff I removed are these settings:
```javascript
	"whitelist_syntaxes": [],
	"blacklist_syntaxes": ["CSS", "SQL", "JavaScript", "Python", "PHP", "JSON"],
 
	/* please use lowercase for the syntax names in the following section: */
	"strip": {
		"php": [
			"<[^>]*>"
		],
		"html": [
			"<[^>]*>"
		]
    }
```
I also rechecked everything on the current `.sublime-settings` file, and everything is present/connected with their implementation on the code, then these settings above are the only think I had removed.


___
> In your version when I open a 70Mb file the count get stuck at 6millons words and does not change when I select stuff.

That was expected because I do not like to keep counting a big file like 70Mb every time. This behavior caused by this line:
```python
g_sleepEvent.wait( Preferences.elapsed_time*100 if Preferences.elapsed_time > 0.01 else 3.0 )
```
It does this:

1. When the last word count time (`elapsed_time`) took **more** than `0.01` seconds, I set the next word count to be run on `elapsed_time * 100` seconds.

1. When the last word count time (`elapsed_time`) took **less** than `0.01` seconds, I set the next word count to be run on `3.0` seconds.

On the case of a 70Mb file, it probably will take more than `0.01` seconds to compute the word counting, then the next time it will count words will be much later. For example, if it took `2.0` seconds, the next word count would be on 200 seconds later.

I did so, to not make my slow computer keep working too much hard for big like that. I think such files are evil and we should not keep peoples computer regularly counting words on such monster files. But if you like, I think I can put that `100` and `0.01` time constants on a configuration file and load it globally. Then people can choose whether their computer keep counting words endlessly or not. 

However, I am not sure about the overall performance implications on the Sublime Text, having a plugin's thread keep counting words regularly on big files like that.

evandrocoan committed:
Removed not used import from os.path import basename

evandrocoan committed:
Fixed AttributeError: 'list' object has no attribute 'split' when

using the setting "enable_count_chars": true.

evandrocoan committed:
Fixed multi lines word count not overriding the initial counting.

evandrocoan committed:
Reimplemented all deprecated features after the overall refactoring

This include the following settings:

// Display the count of words found on current line.
"enable_line_word_count": false,

// Display the count of characters found on current line.
"enable_line_char_count": false,

// An array of syntax names that WordCount should run on.
// Example: ["Plain text", "Markdown"]
// If the array is empty, like it is by default, WordCount will run on any syntax.
"whitelist_syntaxes": [],

// An array of syntax names that WordCount should not run on.
// Example: ["Plain text", "Markdown"]
// If the array is empty, like it is by default, WordCount will run on any syntax.
"blacklist_syntaxes": [],

// Remove regex patterns by syntax. Split portions of text to test later as
// words with a Regular expression. Defaults to String.split() with no
// arguments, means that content will trim() and empty values (all
// whitespaces) are not used. In case of containing some value different
// than empty, the return of "re.findall" will be used.
//
// Please use lowercase for the syntax names in the following section:
"strip": {
    // Example to ignore all tags, including comments, from HTML.
    "html": [
        "<[^>]*>",
    ],
    "php": [
        "<[^>]*>",
    ],
}

evandrocoan committed:
Set WordCount.doCounting() static method as class method.

evandrocoan committed:
Moved WordCount.doCounting(cls) after on_activated_async()

evandrocoan committed:
Fixed the WordCountView.contents attribute being erased after the

first time the word counting is displayed.

evandrocoan commented:
@titoBouzout, I  implemented back every feature. I just did not added a setting to allow the user to change the behavior of this line I mentioned earlier:
```python
g_sleepEvent.wait( Preferences.elapsed_time*100 if Preferences.elapsed_time > 0.01 else 3.0 )
```
Would you like a setting for it?

evandrocoan committed:
Cached wordCountViews[view_id] value on setUpView(cls, view)

evandrocoan committed:
Set dot as thousands separator for status integers and removed

duplicated code under if current_page != 0, on def display().

evandrocoan committed:
Passed view as parameter to updateViewContents(self, view) instead

of caching it from self.view.

evandrocoan committed:
Fixed AttributeError: 'NoneType' object has no attribute 'settings'

Traceback (most recent call last):
  File "F:\SublimeText\sublime_plugin.py", line 538, in on_activated_async
    callback.on_activated_async(v)
  File "F:\SublimeText\Data\Packages\WordCount\WordCount.py", line 143, in on_activated_async
    WordsCount.setUpView( view )
  File "F:\SublimeText\Data\Packages\WordCount\WordCount.py", line 162, in setUpView
    view_settings = view.settings()
AttributeError: 'NoneType' object has no attribute 'settings'

evandrocoan committed:
Fix AttributeError: 'NoneType' object has no attribute 'id'

Traceback (most recent call last):
  File "F:\SublimeText\sublime_plugin.py", line 538, in on_activated_async
    callback.on_activated_async(v)
  File "F:\SublimeText\Data\Packages\WordCount\WordCount.py", line 143, in on_activated_async
    WordsCount.setUpView( view )
  File "F:\SublimeText\Data\Packages\WordCount\WordCount.py", line 167, in setUpView
    view_id = view.id()
